File: Strings/412. Fizz Buzz/412. Fizz Buzz.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        ans = []
        for num in range(1, n + 1):
            div_3 = num % 3
            div_5 = num % 5
            
            if div_3 == 0 and div_5 == 0:
                ans.append("FizzBuzz")
            elif div_3 == 0:
                ans.append("Fizz")
            elif div_5 == 0:
                ans.append("Buzz")
            else:
                ans.append(str(num))
    
        return ans

# Time: O(n)
# Space: O(1)

File: Strings/412. Fizz Buzz/412. Fizz Buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function (n) {
  let ans = [];
  for (let num = 1; num < n + 1; num++) {
    const div_3 = num % 3;
    const div_5 = num % 5;

    if (div_3 === 0 && div_5 === 0) {
      ans.push("FizzBuzz");
    } else if (div_3 == 0) {
      ans.push("Fizz");
    } else if (div_5 == 0) {
      ans.push("Buzz");
    } else {
      ans.push(num.toString());
    }
  }
  return ans;
};

// Time: O(n)
// Space: O(1)
